### PR TITLE
[Unitary hack] Improve docstrings for "task/bloqade"

### PR DESCRIPTION
**Pull request by shubhusion**
Fixes #957 

Improves Docstrings and Remove Markdown errors.

### CODE REVIEW

File: src/bloqade/task/bloqade.py [hunk 1]
```diff
@@ -89,6 +140,15 @@ def run(
 
 @BloqadeTask.set_serializer
 def _serialize(obj: BloqadeTask) -> Dict[str, Any]:
+    """
```
Comment (Roger-luo) [inline src/bloqade/task/bloqade.py:143]: I think these private methods are self-explainable, we don't have to add docstrings to them.

**Review by Roger-luo — CHANGES_REQUESTED** [review]
I'd recommend to finish one of the PRs first instead of submitting a bunch of them to review, so you don't waste any effort and we don't need to review similar issues multiple times. The docstrings in tasks are harder to write as they are deeper in the codebase. I'd recommend you to go through some basic tutorial before going ahead so we make docstring friendly to new comers other than just plain explanation of what the code does.

**Comment by shubhusion** [comment]
> I'd recommend to finish one of the PRs first instead of submitting a bunch of them to review, so you don't waste any effort and we don't need to review similar issues multiple times. The docstrings in tasks are harder to write as they are deeper in the codebase. I'd recommend you to go through some basic tutorial before going ahead so we make docstring friendly to new comers other than just plain explanation of what the code does.

@Roger-luo can you provide a link to tutorials which i can refer to ?

**Comment by Roger-luo** [comment]
https://queracomputing.github.io/bloqade-python-examples/latest/ you can also check the quick start guide in the documentation homepage to understand better what this package does.

**Comment by shubhusion** [comment]
@Roger-luo I have made changes in this PR as requested by you.

File: src/bloqade/task/bloqade.py [hunk 2]
```diff
@@ -13,30 +24,55 @@
...
+
     shots: int
     emulator_ir: EmulatorProgram
     metadata: Dict[str, ParamType]
     compile_cache: Optional[CompileCache] = None
     task_result_ir: Optional[QuEraTaskResults] = None
 
     def _geometry(self) -> Geometry:
+        """
```
Comment (Roger-luo) [inline src/bloqade/task/bloqade.py:51]: this is a private function

File: src/bloqade/task/bloqade.py [hunk 3]
```diff
@@ -47,7 +83,22 @@ def run(
         nsteps: int = 2_147_483_647,
         interaction_picture: bool = False,
     ) -> "BloqadeTask":
-
+        """
```
Comment (Roger-luo) [inline src/bloqade/task/bloqade.py:86]: please also add one example to demonstrate how this function can be used. This is a user API.

**Review by Roger-luo — CHANGES_REQUESTED** [review]
Please try to finish one PR before requesting reviews on multiple ones, otherwise this will result in duplicated review comments that are hard to manage.

**Pull request closed without merging**